app/models/minion.rb

```diff
@@ -1,4 +1,25 @@
 # frozen_string_literal: true
+
+require 'pharos/salt'
+
 # Minion represents the minions that have been registered in this application.
 class Minion < ApplicationRecord
+
+  def assign_role(available_roles)
```

Comment:
I would like it better if this method took only one role and assigned that. You can move the random role selection outside of this method.